buildenvs/base.Dockerfile

```diff
@@ -0,0 +1,143 @@
+# COPY --from=gcc-arm64 /libexec/gcc/aarch64-linux-gnu/12.2.0/collect2 /libexec/gcc/aarch64-linux-gnu/12.2.0/collect2
+# COPY the others if needed
+
+# Link the GCC toolchain
+RUN set -xe; \
+    ln -s /bin/${GCC_PREFIX}-as         /bin/as; \
+    ln -s /bin/${GCC_PREFIX}-ar         /bin/ar; \
+    ln -s /bin/${GCC_PREFIX}-c++        /bin/c++; \
+    ln -s /bin/${GCC_PREFIX}-c++filt    /bin/c++filt; \
+    ln -s /bin/${GCC_PREFIX}-elfedit    /bin/elfedit; \
+    ln -s /bin/${GCC_PREFIX}-gcc        /bin/cc; \
+    ln -s /bin/${GCC_PREFIX}-gcc        /bin/gcc; \
+    ln -s /bin/${GCC_PREFIX}-gcc-ar     /bin/gcc-ar; \
+    ln -s /bin/${GCC_PREFIX}-gcc-nm     /bin/gcc-nm; \
+    ln -s /bin/${GCC_PREFIX}-gcc-ranlib /bin/gcc-ranlib; \
+    ln -s /bin/${GCC_PREFIX}-gccgo      /bin/gccgo; \
+    ln -s /bin/${GCC_PREFIX}-gcov       /bin/gcov; \
+    ln -s /bin/${GCC_PREFIX}-gcov-dump  /bin/gcov-dump; \
+    ln -s /bin/${GCC_PREFIX}-gcov-tool  /bin/gcov-tool; \
+    ln -s /bin/${GCC_PREFIX}-gprof      /bin/gprof; \
+    ln -s /bin/${GCC_PREFIX}-ld         /bin/ld; \
+    ln -s /bin/${GCC_PREFIX}-nm         /bin/nm; \
+    ln -s /bin/${GCC_PREFIX}-objcopy    /bin/objcopy; \
+    ln -s /bin/${GCC_PREFIX}-objdump    /bin/objdump; \
+    ln -s /bin/${GCC_PREFIX}-ranlib     /bin/ranlib; \
+    ln -s /bin/${GCC_PREFIX}-readelf    /bin/readelf; \
+    ln -s /bin/${GCC_PREFIX}-size       /bin/size; \
+    ln -s /bin/${GCC_PREFIX}-strings    /bin/strings; \
+    ln -s /bin/${GCC_PREFIX}-strip      /bin/strip;
+
+# Copy the QEMU emulator
+COPY --from=qemu /bin/                  /bin
+COPY --from=qemu /share/qemu/           /share/qemu
+COPY --from=qemu /lib/x86_64-linux-gnu/ /lib/x86_64-linux-gnu
+
+# COPY --from=qemu /lib/arm-linux-gnueabihf/ /lib/arm-linux-gnueabihf
+# COPY --from=qemu /lib/aarch64-linux-gnu/ /lib/aarch64-linux-gnu
+
+# Unikraft workdir and environment
+WORKDIR /usr/src/unikraft/apps/app
+
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+# Install unikraft dependencies
+RUN set -xe; \
+    apt-get update; \
+    apt-get install -y --no-install-recommends \
+      make=4.3-4.1 \
+      libncursesw5-dev=6.2+20201114-2 \
+      libncursesw5=6.2+20201114-2 \
+      libyaml-dev=0.2.2-1 \
+      flex=2.6.4-8 \
+      git=1:2.30.2-1 \
+      wget=1.21-1+deb11u1 \
+      patch=2.7.6-7 \
+      gawk=1:5.1.0-1 \
+      socat=1.7.4.1-3 \
+      bison=2:3.7.5+dfsg-1 \
+      bindgen=0.55.1-3+b1 \
+      bzip2=1.0.8-4 \
+      unzip=6.0-26+deb11u1 \
+      uuid-runtime=2.36.1-8+deb11u1 \
+      openssh-client=1:8.4p1-5+deb11u1 \
+      autoconf=2.69-14 \
+      xz-utils=5.2.5-2.1~deb11u1 \
+      python3=3.9.2-3 \
+      libgit2-dev=1.1.0+dfsg.1-4 \
+      ca-certificates=20210119; \
+    apt-get clean; \
+    rm -Rf /var/cache/apt/*; \
+    rm -Rf /var/lib/apt/lists/*; \
+    mkdir -p /usr/src/unikraft/unikraft; \
+    mkdir -p /usr/src/unikraft/libs; \
+    mkdir -p /usr/src/unikraft/apps/app; \
```

Comment:
Actually we can remove these since these are an artefact from `pykraft`:
```diff
-     mkdir -p /usr/src/unikraft/unikraft; \
-     mkdir -p /usr/src/unikraft/libs; \
-     mkdir -p /usr/src/unikraft/apps/app; \
```